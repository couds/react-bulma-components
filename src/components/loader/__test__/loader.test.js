import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '..';

describe('Box component', () => {
  it('Should Exist', () => {
    expect(Loader).toMatchSnapshot();
  });
  it('Should have box classname', () => {
    const component = renderer.create(<Loader />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat Bulma class with classes in props', () => {
    const component = renderer.create(<Loader className="other-class test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should render as an html section', () => {
    const component = renderer.create(<Loader renderAs="section" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should have custom inline styles', () => {
    const component = renderer.create(<Loader renderAs="section" style={{ width: 200, zIndex: 1 }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
