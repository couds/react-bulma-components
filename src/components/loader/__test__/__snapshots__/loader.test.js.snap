// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Should Exist 1`] = `
Object {
  "$$typeof": Symbol(react.forward_ref),
  "defaultProps": Object {
    "backgroundColor": null,
    "className": "",
    "clearfix": false,
    "clipped": false,
    "hidden": false,
    "invisible": false,
    "italic": false,
    "marginless": false,
    "overlay": false,
    "paddingless": false,
    "pull": null,
    "radiusless": false,
    "renderAs": "div",
    "responsive": undefined,
    "shadowless": false,
    "style": Object {},
    "textAlignment": null,
    "textColor": null,
    "textSize": null,
    "textTransform": null,
    "textWeight": null,
    "unselectable": false,
  },
  "propTypes": Object {
    "backgroundColor": [Function],
    "className": [Function],
    "clearfix": [Function],
    "clipped": [Function],
    "hidden": [Function],
    "invisible": [Function],
    "italic": [Function],
    "marginless": [Function],
    "onClick": [Function],
    "overlay": [Function],
    "paddingless": [Function],
    "pull": [Function],
    "radiusless": [Function],
    "renderAs": [Function],
    "responsive": [Function],
    "shadowless": [Function],
    "style": [Function],
    "textAlignment": [Function],
    "textColor": [Function],
    "textSize": [Function],
    "textTransform": [Function],
    "textWeight": [Function],
    "unselectable": [Function],
  },
  "render": [Function],
}
`;

exports[`Loader component Should concat Bulma class with classes in props 1`] = `
<div
  className="loader other-class test"
  style={Object {}}
>
  <div
    className="loader-background"
  />
  <div
    className="loader-icon"
  />
</div>
`;

exports[`Loader component Should have box classname 1`] = `
<div
  className="loader"
  style={Object {}}
>
  <div
    className="loader-background"
  />
  <div
    className="loader-icon"
  />
</div>
`;

exports[`Loader component Should have custom inline styles 1`] = `
<section
  className="loader"
  style={
    Object {
      "width": 200,
      "zIndex": 1,
    }
  }
>
  <div
    className="loader-background"
  />
  <div
    className="loader-icon"
  />
</section>
`;

exports[`Loader component Should render as an html section 1`] = `
<section
  className="loader"
  style={Object {}}
>
  <div
    className="loader-background"
  />
  <div
    className="loader-icon"
  />
</section>
`;
